crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -230,6 +230,14 @@ struct UncommittedData {
     // table as they are flushed to the db.
     pending_transaction_writes: DashMap<TransactionDigest, Arc<TransactionOutputs>>,
 
+    // Transactions outputs from Mysticeti fastpath certified transactions.
+    // These outputs are not written to pending_transaction_writes until we are sure
+    // that they will not get rejected by consensus.
+    // TODO(fastpath): We need to find a way to remove entries that did not get
+    // finalized. We could use rejection status from mystticeti, but it is
+    // not 100% reliable due to data races.
```

Comment:
Maybe another option is to just make this a bounded-sized cache. It seems that nothing bad happens if outputs get evicted before the transaction is finalized, we just re-execute, right?

The only case to be concerned about is equivocation over effects (in case we get different effects upon re-execution due to a bug), but I believe the existing equivocation prevention logic should be enough.

Reply:
The existing equivocation prevention logic will not be able to cover this case, because we no longer sign over effects.
In fact, regardless of how we handle the eviction here, we won't be able to do the same kind of equivocation prevention across crash recoveries, because we do not always persist effects returned through mysticeti fastpath.
If we believe this is important to handle, we will need to add a new table in epoch store that maps from tx digest to returned effects.